org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/config/PropertiesHelperTest.java

```diff
@@ -899,36 +784,126 @@ else if ("dbSystemProperties".equals(configUrl)) {
-      m_throwExceptionOnRead = throwExceptionOnRead;
-    }
+  @Test
+  public void testParseJsonSingleIntegerArgument() {
+    assertThrows(RuntimeException.class, () -> testParseJsonInternal("1"));
+  }
+
+  @Test
+  public void testParseJsonSingleQuotedStringArgument() {
+    assertThrows(RuntimeException.class, () -> testParseJsonInternal("\"a\""));
+  }
+
+  @Test
+  public void testParseJsonReadEmptyJsonObjectString() {
+    assertEquals(Collections.emptyMap(), testParseJsonInternal("{}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithSingleEmptyStringProperty() {
+    assertEquals(Map.of("testKey", ""), testParseJsonInternal("{\"testKey\": \"\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithSingleStringProperty() {
+    assertEquals(Map.of("testKey", "testValue"), testParseJsonInternal("{\"testKey\": \"testValue\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithMultipleStringProperties() {
+    assertEquals(Map.of(
+            "testKey1", "testValue1",
+            "testKey2", "testValue2"),
+        testParseJsonInternal("{\"testKey1\": \"testValue1\", \"testKey2\": \"testValue2\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithPropertyWithNullValue() {
+    assertEquals(CollectionUtility.hashMap( // Map.of cannot be used with a null value
+            ImmutablePair.of("testKey1", null),
+            ImmutablePair.of("testKey2", "testValue2")),
+        testParseJsonInternal("{\"testKey1\": null, \"testKey2\": \"testValue2\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithIntegerProperty() {
+    assertEquals(Map.of(
+            "testKey1", "1",
+            "testKey2", "testValue2"),
+        testParseJsonInternal("{\"testKey1\": 1, \"testKey2\": \"testValue2\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithBooleanProperty() {
+    assertEquals(Map.of(
+            "testKey1", "true",
+            "testKey2", "testValue2"),
+        testParseJsonInternal("{\"testKey1\": true, \"testKey2\": \"testValue2\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithDoubleProperty() {
+    assertEquals(Map.of(
+            "testKey1", "1.2",
+            "testKey2", "testValue2"),
+        testParseJsonInternal("{\"testKey1\": 1.2, \"testKey2\": \"testValue2\"}"));
+  }
+
+  @Test
+  public void testParseJsonReadJsonObjectWithJsonStringProperty() {
+    assertEquals(Map.of(
+            "testKey1", "{\"subKey1\":\"subValue1\",\"subKey2\":\"subValue2\"}",
+            "testKey2", "[\"testValue2Sub1\",\"testValue2Sub2\",\"testValue2Sub3\"]"),
+        testParseJsonInternal("{\"testKey1\": {\"subKey1\": \"subValue1\", \"subKey2\": \"subValue2\"}, \"testKey2\": [\"testValue2Sub1\", \"testValue2Sub2\", \"testValue2Sub3\"]}"));
+  }
+
+  protected Map<String, String> testParseJsonInternal(String propertyValue) {
+    return new PropertiesHelper(Mockito.mock(IPropertyProvider.class)).parseJson(propertyValue);
```

Comment:
Why not just use new PropertiesHelper(null).parseJson(propertyValue)? No reason to use a mock here